### ISSUE TITLE
Nil pointer when setting basic auth on frontend

### ISSUE THREAD

**Issue by napstr** [containous/traefik#4211]
### Do you want to request a *feature* or report a *bug*?

bug

### What did you do?

Set up basic auth on a frontend via docker provider. See docker-compose file further below for full config to reproduce this.
Then visited the URL in the browser.

### What did you expect to see?
An authentication prompt.

### What did you see instead?
Internal Server Error

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.4
Codename:     maroilles
Go version:   go1.11.1
Built:        2018-10-30_10:44:30AM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

I'm running traefik via docker-compose (see further below for the file) on an Ubuntu 16.04 machine.

(this output, which I got by `docker exec traefik traefik bug` seems to be wrong btw, see the docker-compose file where I'm actually enabled anonymous data collection and setting debug to true etc)

```json
{
 "LifeCycle": {
  "RequestAcceptGraceTimeout": 0,
  "GraceTimeOut": 0
 },
 "GraceTimeOut": 0,
 "Debug": false,
 "CheckNewVersion": true,
 "SendAnonymousUsage": false,
 "AccessLogsFile": "",
 "AccessLog": null,
 "TraefikLogsFile": "",
 "TraefikLog": null,
 "Tracing": null,
 "LogLevel": "",
 "EntryPoints": {},
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [
  "http"
 ],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 0,
 "InsecureSkipVerify": false,
 "RootCAs": null,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "RespondingTimeouts": null,
 "ForwardingTimeouts": null,
 "AllowMinWeightZero": false,
 "KeepTrailingSlash": false,
 "Web": null,
 "Docker": null,
 "File": null,
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ServiceFabric": null,
 "Rest": null,
 "API": null,
 "Metrics": null,
 "Ping": null,
 "HostResolver": null,
 "ConfigFile": ""
}
```
This docker-compose file should make reproducing the issue easy:

```yaml
version: '3'

networks:
  traefik:
    external: true

services:

  traefik:
    image: traefik:v1.7-alpine
    container_name: traefik
    restart: always
    ports:
      - 80:80
    networks:
      - traefik
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    command:
      - --debug=true
      - --logLevel=DEBUG
      - --sendAnonymousUsage=true
      - --defaultEntryPoints=http
      - "--entryPoints=Name:http Address::80"
      - --docker.domain=${DOMAIN}
      - --docker.endpoint=unix:///var/run/docker.sock
      - --docker.watch=true
      - --docker.exposedByDefault=false

  whoami:
    image: emilevauge/whoami
    restart: always
    networks:
      - traefik
    expose:
      - 80
    labels:
      - "traefik.docker.network=traefik"
      - "traefik.enable=true"
      - "traefik.port=80"
      - "traefik.backend=foo"
      - "traefik.frontend.rule=Host:${DOMAIN}"
      - "traefik.frontend.auth.based.users=${BASIC_AUTH}"
```

with `DOMAIN=dont-do-this-at.work` and `BASIC_AUTH=metrics:$apr1$AOc2P0ia$afplbiVCzpfaLkwt816FK.` (the pass is 'whatever42' if you decided to use this hash, or just make your own with `htpasswd`. I've tried MD5 and bcrypt hashes, both produce the error) in my environment, for example in a `.env` file next to the docker-compose file.

### If applicable, please paste the log output at DEBUG level (`--logLevel=DEBUG` switch)

```
$ docker-compose up
Starting traefik                   ... done
Creating npe_whoami_1_f1015354c027 ... done
Attaching to traefik, npe_whoami_1_41cd7db5169a
whoami_1_41cd7db5169a | Starting up on port 80
traefik    | time="2018-11-19T22:42:15Z" level=info msg="Traefik version v1.7.4 built on 2018-10-30_10:44:30AM"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Global configuration loaded {\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"GraceTimeOut\":0,\"Debug\":true,\"CheckNewVersion\":true,\"SendAnonymousUsage\":true,\"AccessLogsFile\":\"\",\"AccessLog\":null,\"TraefikLogsFile\":\"\",\"TraefikLog\":null,\"Tracing\":null,\"LogLevel\":\"DEBUG\",\"EntryPoints\":{\"http\":{\"Address\":\":80\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}}},\"Cluster\":null,\"Constraints\":[],\"ACME\":null,\"DefaultEntryPoints\":[\"http\"],\"ProvidersThrottleDuration\":2000000000,\"MaxIdleConnsPerHost\":200,\"IdleTimeout\":0,\"InsecureSkipVerify\":false,\"RootCAs\":null,\"Retry\":null,\"HealthCheck\":{\"Interval\":30000000000},\"RespondingTimeouts\":null,\"ForwardingTimeouts\":null,\"AllowMinWeightZero\":false,\"KeepTrailingSlash\":false,\"Web\":null,\"Docker\":{\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"unix:///var/run/docker.sock\",\"Domain\":\"dont-do-this-at.work\",\"TLS\":null,\"ExposedByDefault\":false,\"UseBindPortIP\":false,\"SwarmMode\":false,\"Network\":\"\"},\"File\":null,\"Marathon\":null,\"Consul\":null,\"ConsulCatalog\":null,\"Etcd\":null,\"Zookeeper\":null,\"Boltdb\":null,\"Kubernetes\":null,\"Mesos\":null,\"Eureka\":null,\"ECS\":null,\"Rancher\":null,\"DynamoDB\":null,\"ServiceFabric\":null,\"Rest\":null,\"API\":null,\"Metrics\":null,\"Ping\":null,\"HostResolver\":null}"
traefik    | time="2018-11-19T22:42:15Z" level=info msg="\nStats collection is enabled.\nMany thanks for contributing to Traefik's improvement by allowing us to receive anonymous information from your configuration.\nHelp us improve Traefik by leaving this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
traefik    | time="2018-11-19T22:42:15Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc0004c59e0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
traefik    | time="2018-11-19T22:42:15Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
traefik    | time="2018-11-19T22:42:15Z" level=info msg="Starting server on :80"
traefik    | time="2018-11-19T22:42:15Z" level=info msg="Starting provider *docker.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"unix:///var/run/docker.sock\",\"Domain\":\"dont-do-this-at.work\",\"TLS\":null,\"ExposedByDefault\":false,\"UseBindPortIP\":false,\"SwarmMode\":false,\"Network\":\"\"}"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Provider connection established with docker 18.09.0 (API 1.39)"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="originLabelsmap[com.docker.compose.container-number:1 com.docker.compose.oneoff:False com.docker.compose.service:whoami com.docker.compose.slug:41cd7db5169afaa027632e92e1fa2d88ed8bd49b217b9cd3fa48632ca45b8c com.docker.compose.version:1.23.1 traefik.backend:foo traefik.docker.network:traefik com.docker.compose.config-hash:1c459ff6e8cc9a224c32794db66ed8f2e4f58a0a8d6a3202b01a7beae28aa20c traefik.frontend.rule:Host:dont-do-this-at.work traefik.enable:true traefik.frontend.auth.based.users:metrics:$apr1$AOc2P0ia$afplbiVCzpfaLkwt816FK. traefik.port:80 com.docker.compose.project:npe]"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="allLabelsmap[:map[traefik.frontend.auth.based.users:metrics:$apr1$AOc2P0ia$afplbiVCzpfaLkwt816FK. traefik.port:80 traefik.docker.network:traefik traefik.backend:foo traefik.enable:true traefik.frontend.rule:Host:dont-do-this-at.work]]"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Filtering disabled container /traefik"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="originLabelsmap[com.docker.compose.project:npe traefik.frontend.auth.based.users:metrics:$apr1$AOc2P0ia$afplbiVCzpfaLkwt816FK. traefik.port:80 traefik.backend:foo traefik.docker.network:traefik com.docker.compose.config-hash:1c459ff6e8cc9a224c32794db66ed8f2e4f58a0a8d6a3202b01a7beae28aa20c com.docker.compose.container-number:1 com.docker.compose.oneoff:False com.docker.compose.service:whoami com.docker.compose.slug:41cd7db5169afaa027632e92e1fa2d88ed8bd49b217b9cd3fa48632ca45b8c com.docker.compose.version:1.23.1 traefik.enable:true traefik.frontend.rule:Host:dont-do-this-at.work]"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="allLabelsmap[:map[traefik.frontend.rule:Host:dont-do-this-at.work traefik.frontend.auth.based.users:metrics:$apr1$AOc2P0ia$afplbiVCzpfaLkwt816FK. traefik.port:80 traefik.backend:foo traefik.docker.network:traefik traefik.enable:true]]"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Backend backend-foo: no load-balancer defined, fallback to 'wrr' method"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Configuration received from provider docker: {\"backends\":{\"backend-foo\":{\"servers\":{\"server-npe-whoami-1-41cd7db5169a-4dbdc7fb202a58a390c42b961e83dde8\":{\"url\":\"http://172.26.0.3:80\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"frontend-Host-dont-do-this-at-work-0\":{\"entryPoints\":[\"http\"],\"backend\":\"backend-foo\",\"routes\":{\"route-frontend-Host-dont-do-this-at-work-0\":{\"rule\":\"Host:dont-do-this-at.work\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null,\"auth\":{}}}}"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Wiring frontend frontend-Host-dont-do-this-at-work-0 to entryPoint http"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Creating backend backend-foo"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-dont-do-this-at-work-0"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Creating load-balancer wrr"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Creating server server-npe-whoami-1-41cd7db5169a-4dbdc7fb202a58a390c42b961e83dde8 at http://172.26.0.3:80 with weight 1"
traefik    | time="2018-11-19T22:42:15Z" level=debug msg="Creating route route-frontend-Host-dont-do-this-at-work-0 Host:dont-do-this-at.work"
traefik    | time="2018-11-19T22:42:15Z" level=info msg="Server configuration reloaded on :80"
traefik    | time="2018-11-19T22:42:22Z" level=error msg="Recovered from panic in HTTP handler [<snip> - /b]: runtime error: invalid memory address or nil pointer dereference"
traefik    | time="2018-11-19T22:42:22Z" level=error msg="Stack: goroutine 36 [running]:\ngithub.com/containous/traefik/middlewares.recoverFunc(0x7f0ebb409540, 0xc00000c978, 0xc000117200)\n\t/go/src/github.com/containous/traefik/middlewares/recover.go:40 +0x1b7\npanic(0x21e5ca0, 0x4551d40)\n\t/usr/local/go/src/runtime/panic.go:513 +0x1b9\ngithub.com/containous/traefik/middlewares/auth.(*Authenticator).ServeHTTP(0xc0003ed820, 0x7f0ebb409540, 0xc00000c980, 0xc000117800, 0xc0003eddc0)\n\t/go/src/github.com/containous/traefik/middlewares/auth/authenticator.go:161 +0x29\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x2949220, 0xc0003ed820, 0xc0003edae0, 0x7f0ebb409540, 0xc00000c980, 0xc000117800)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x9c\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.(*Negroni).ServeHTTP(0xc00018e060, 0x7f0ebb409540, 0xc00000c978, 0xc000117800)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:81 +0xee\ngithub.com/containous/traefik/vendor/github.com/containous/mux.(*Router).ServeHTTP(0xc0006708a0, 0x7f0ebb409540, 0xc00000c978, 0xc000117800)\n\t/go/src/github.com/containous/traefik/vendor/github.com/containous/mux/mux.go:133 +0xf1\ngithub.com/containous/traefik/middlewares.(*HandlerSwitcher).ServeHTTP(0xc0003055c0, 0x7f0ebb409540, 0xc00000c978, 0xc000117300)\n\t/go/src/github.com/containous/traefik/middlewares/handlerSwitcher.go:24 +0x6f\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.Wrap.func1(0x7f0ebb409540, 0xc00000c978, 0xc000117300, 0xc0003edda0)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:41 +0x4d\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.HandlerFunc.ServeHTTP(0xc00054b9a0, 0x7f0ebb409540, 0xc00000c978, 0xc000117300, 0xc0003edda0)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:24 +0x4e\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x294dee0, 0xc00054b9a0, 0xc00054ba00, 0x7f0ebb409540, 0xc00000c978, 0xc000117300)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x9c\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP-fm(0x7f0ebb409540, 0xc00000c978, 0xc000117300)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x60\nnet/http.HandlerFunc.ServeHTTP(0xc0003edd80, 0x7f0ebb409540, 0xc00000c978, 0xc000117300)\n\t/usr/local/go/src/net/http/server.go:1964 +0x44\ngithub.com/containous/traefik/middlewares.(*RequestHost).ServeHTTP(0x4592118, 0x7f0ebb409540, 0xc00000c978, 0xc000117200, 0xc0003edd80)\n\t/go/src/github.com/containous/traefik/middlewares/request_host.go:20 +0x1ec\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x2949020, 0x4592118, 0xc00054b9e0, 0x7f0ebb409540, 0xc00000c978, 0xc000117200)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x9c\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP-fm(0x7f0ebb409540, 0xc00000c978, 0xc000117200)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x60\nnet/http.HandlerFunc.ServeHTTP(0xc0003edd60, 0x7f0ebb409540, 0xc00000c978, 0xc000117200)\n\t/usr/local/go/src/net/http/server.go:1964 +0x44\ngithub.com/containous/traefik/middlewares.NegroniRecoverHandler.func1(0x7f0ebb409540, 0xc00000c978, 0xc000117200, 0xc0003edd60)\n\t/go/src/github.com/containous/traefik/middlewares/recover.go:24 +0x87\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.HandlerFunc.ServeHTTP(0x27143e0, 0x7f0ebb409540, 0xc00000c978, 0xc000117200, 0xc0003edd60)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:24 +0x4e\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x294dee0, 0x27143e0, 0xc00054b9c0, 0x7f0ebb409540, 0xc00000c978, 0xc000117200)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x9c\ngithub.com/containous/traefik/vendor/github.com/urfave/negroni.(*Negroni).ServeHTTP(0xc00058a030, 0x296fe60, 0xc00017ca80, 0xc000117200)\n\t/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:81 +0xee\ngithub.com/containous/traefik/vendor/github.com/containous/mux.(*Router).ServeHTTP(0xc000397d40, 0x296fe60, 0xc00017ca80, 0xc000117200)\n\t/go/src/github.com/containous/traefik/vendor/github.com/containous/mux/mux.go:133 +0xf1\ngithub.com/containous/traefik/h2c.Server.Serve.func1(0x296fe60, 0xc00017ca80, 0xc000116f00)\n\t/go/src/github.com/containous/traefik/h2c/h2c.go:78 +0x409\nnet/http.HandlerFunc.ServeHTTP(0xc00054bc00, 0x296fe60, 0xc00017ca80, 0xc000116f00)\n\t/usr/local/go/src/net/http/server.go:1964 +0x44\nnet/http.serverHandler.ServeHTTP(0xc00002b1e0, 0x296fe60, 0xc00017ca80, 0xc000116f00)\n\t/usr/local/go/src/net/http/server.go:2741 +0xab\nnet/http.(*conn).serve(0xc0002d1360, 0x2972360, 0xc0004d3500)\n\t/usr/local/go/src/net/http/server.go:1847 +0x646\ncreated by net/http.(*Server).Serve\n\t/usr/local/go/src/net/http/server.go:2851 +0x2f5\n"

```

**Comment by napstr** [comment]
Alright, looks like this was just a typo on my end:
```diff
- - "traefik.frontend.auth.based.users=${BASIC_AUTH}"
+ - "traefik.frontend.auth.basic.users=${BASIC_AUTH}"
```
Works fine now. Weird error to throw on an invalid label but ¯\\\_(ツ)_/¯

**Comment by traefiker** [comment]
Closed by #4212.
